### ISSUE TITLE
Oil Refinery

### ISSUE THREAD

**Issue by mensreaMC** [mensreaMC/Deus-Ex-Machina#24]
# INTRO

The backbone of a hydrocarbon infrastructure is the refinery. Through a variety of chemical and thermal means the refinery is capable of breaking down oils into their various useful components. Though Light Sweet crude can be burned by itself, it is usually more efficient to turn it into the fuels and components which modern industry relies on to function. The refinery itself is a monolithic construct with a huge footprint, energy requirement and requisite capitol investment. Real life refineries are billion dollar projects that must be ran 24/7, as to start them up would take days and involve a series of rigorous adjustments that only trained, experienced engineers can perform.
# DETAILS

I.refinery
A.facility
1.configuration and capacity
a.consists of a control block, a motor block, mechanism blocks, component blocks, casing blocks, pipe blocks and condenser blocks
b."Refinery control block" scans a 40x40x20 area for the blocks when placed and before each operation or it doesnt place/run
b.only ever needs one control block which acts as the GUI and the the output/input/power receiver
c.other blocks in varying amounts determine refineries capacity (smaller refineries have less power and capitol requirements but lower potential output and efficiency)
d.the minimum blocks it needs are 20 component blocks, 20 casing blocks, 10 pipe blocks, 10 condensor blocks, 6 mechanism blocks and 1 motor block rated no less than 100 kw. with the minimum configuration the refinery would have a size value of 1
e.each time the minimum number of blocks is multiplied by a number divisible by 2 the size factor goes up by 1
-example: the player uses 4 times the minimum number of required blocks, making the size factor 4
f.size values determine the capacity and the efficiency
-for each size factor of 1 the refinery can process 100 barrels at once
-example: the players size value of 4 refinery can process up to 400 barrels at once
g.refineries process petroleum in batches of 100 barrels at a time
h.the appropriate number of empty barrels must be placed in the input slot or else it will not process it. also, the barrels of petroleum which will be emptied in the process will be included in the calculation
i.players must use motor blocks in the 26-100 kw range. motor blocks cannot be mixed, despite rating
-example: the player tries to use a motor block rated for 2 mw on their city sized refinery, but the refinery will not operate. the player would have to swap the 2 mw motor for 20 26-100 kw motors for the refinery to function.

2.power and time requirements
a.4.4 mwh per 100 barrels of light sweet crude; takes 60 minutes
b.5.3 mwh per 100 barrels of heavy sour crude; takes 60 minutes

3.water requirements
4.heat-up time

5.inputs
-note: EROI is a complex subject based off of gross supply, total energy inputs to get a given output, and other factors such as transportation, capitol investments, etc. however, we will simplify things and set the values at a fixed rate. in short, our EROI's will be a reflection of the energy used to pump the oil out of the ground and refine it VS the total amount of energy that would be gained from using it to generate power
-note: i will also design it so light sweet produces much more gas than heavy sour, which will produce more tar. both are arguably useful products
-note: though barrels hold 42 gallons total yield of products is 45 gallons, due to inputs of water and energy
-note: the names of oil are based off their sulfur content and viscosity, with light sweet having low sulfur and containing more valuable fuels while heavy contains more tar

a.light sweet crude
-EROI of 30:1
-42 gallon barrel produces: 19 gallons of gasoline, 11 gallons of diesel, 7 gallons of LPG, 4 gallons of kerosene, 2 gallons of tar, 1 gallon of lubricating oil, 1 unit of coke, 1 unit of sulfur
-1702 kw/barrel
-54 kwh to extract and refine 1 barrel (10 to pump it, 44 to refine it)
-100 barrels: 1900 gallons of gas (44 barrels), 1100 gallons of diesel (26 barrels), 700 gallons of LPG (16 tanks), 400 gallons of kerosene (9 barrels), 200 gallons of tar (5 barrels), 2 barrels of lubricating oil, 100 units of coke, 100 units of sulfur
-170,200 kwh per 100 barrels
-5,444 kwh to extract and refine 100 barrels (1000 to pump it, 4400 to refine it)

b.heavy sour crude
-EROI 15:1
-42 gallon barrel produces: 15 gallons of gasoline, 9 gallons of diesel, 7 gallons of tar, 4 gallons of kerosene, 2 gallons of LPG, 1 gallon of lubricating oil, 1 unit of coke, 7 units of sulfur
-1,033 kwh per barrel
-68 kwh to extract and refine 1 barrel (15 to pump it, 53 to refine it)
-100 barrels: 1500 gallons of gas (35 barrels), 900 gallons of diesel (21 barrels), 700 gallons of tar (16 barrels), 400 gallons of kerosene (9 barrels), 200 gallons of LPG (5 tanks), 2 barrels of lubricating oil, 100 units of coke, 700 units of sulfur
-103,000 kwh per 100 barrels
-6,800 kwh to extract and refine 100 barrels (1500 to pump it, 5300 to refine it)

B.products, by weight
1.LPG
-Liquified Petroleum Gas
-propane, butane, or a mix of the two
-"i sell propane and propane accessories"
2.ethylene
-a light hydrocarbon gas responsible for the majority of the worlds plastics
-instead of adding it as a product we should let the player further distill LPG
3.kerosene
-jet fuel 
-lamp oil
-can be used in some motors
-more stable than gasoline
4.gasoline
-cause of the iraq war
-most energy dense form of fuel by weight
-perfect for ICU's in portable tools
5.diesel
-named for vin diesel, second most energy dense form of fuel and source of "fast & furious"
-used in very efficient but much larger ICE's
6.lubricating oils
-self explanatory
7.tar
-used mostly in construction for shingles, asphalt, water proofing
-used to coat tax collecters who can then be effectively feathered
8.coke
-basically heavy residual tars heated to the point they become almost pure carbon
9.sulfur
-elemental sulfur can be extracted from petrochemicals in commercially significant quantities and is used for many things in industry

**Comment by mensreaMC** [comment]
REVISION 1
-clarified which sort of motor must be used

**Comment by mensreaMC** [comment]
REVISION 2
-updated according to fuel index values
-+5 eroi to light sweet crude
